pkg/apis/v1alpha1/types.go

```diff
@@ -223,5 +241,10 @@ func compareDeployments(a, b *appsv1.Deployment) bool {
 			}
 		}
 	}
+
+	if b.Spec.Replicas != a.Spec.Replicas {
```

Comment:
Curious if this is comparing values or addresses

Reply:
Double checked it and it compares addresses, good catch. Fixed.